runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractQueuedStreamView.java

```diff
@@ -580,6 +584,8 @@ public synchronized ILogData current() {
         if (Address.nonAddress(context.getGlobalPointer())) {
             return null;
         }
+
+        // The ILogData returned by current() would be used for rollback.
         return read(context.getGlobalPointer());
```

Comment:
Why you are not checking compaction mark here?

Reply:
I added the compaction mark check here.